### PR TITLE
test: refactored tests for better async resilience and patterns

### PR DESCRIPTION
**Pull request by cerebrl**
Abstracted out repeated patterns or confusing patterns. Aligned all element selectors and element
state assertions to match standards.

I will add to our documentation more information about why I made these changes in the future. Right now, I just want to get this code change complete.

### CODE REVIEW

File: tests/widget/modal/widget-modal.us_es.test.js [hunk 1]
```diff
@@ -1,19 +1,16 @@
...
+test.use({ locale: 'es-US' });
+test('modal widget in US Spanish', async ({ page }) => {
   await page.goto('widget/modal');
+
   const loginButton = page.locator('button', { hasText: 'Open Login Modal' });
-  const dialog = page.locator('dialog');
-  expect(await dialog.isVisible()).toBeFalsy();
 
   await loginButton.click();
-  expect(await dialog.isVisible()).toBeTruthy();
```
Comment (cerebrl) [inline tests/widget/modal/widget-modal.us_es.test.js:13]: These `isVisible` assertions are just a repeat from the base modal test, so removing them for efficiency.

File: tests/widget/modal/widget-modal.test.js [hunk 2]
```diff
@@ -2,11 +2,14 @@ import { expect, test } from '@playwright/test';
...
 test('modal widget', async ({ page }) => {
   await page.goto('widget/modal');
+
   const loginButton = page.locator('button', { hasText: 'Open Login Modal' });
   const dialog = page.locator('dialog');
   expect(await dialog.isVisible()).toBeFalsy();
 
   await loginButton.click();
+  // Add just a bit of a delay to ensure dialog responds
+  await page.waitForTimeout(1000);
```
Comment (cerebrl) [inline tests/widget/modal/widget-modal.test.js:12]: I had a random failure here. It seems the network response was a bit slow and the dialog didn't have enough time to open, so added just a slight delay.

File: tests/widget/inline/widget-inline.ca_fr.test.js [hunk 3]
```diff
@@ -1,9 +1,7 @@
 import { expect, test } from '@playwright/test';
 
-test('inline widget in Canadian French', async ({ browser }) => {
-  const context = await browser.newContext({ locale: 'fr-CA' });
-  const page = await context.newPage();
-
+test.use({ locale: 'fr-CA' });
+test('inline widget in Canadian French', async ({ page }) => {
```
Comment (cerebrl) [inline tests/widget/inline/widget-inline.ca_fr.test.js:4]: I think this pattern is a bit more concise and keeps it consistent with the rest of the tests.

**Review by ryanbas21 — APPROVED** [review]

File: tests/widget/inline/widget-inline.es.test.js [hunk 4]
```diff
@@ -1,14 +1,19 @@
...
-  await page.fill('text=Contraseña', 'j56eKtae*1');
-  await page.locator('button', { hasText: 'Iniciar sesion' }).click();
+  const username = page.getByRole('textbox', { name: "Nombre de usuario" });
+
+  await username.fill('demouser')
+
+  const password = page.getByRole('textbox', { name: "Contraseña" });
+  await password.fill('j56eKtae*1');
+
+  const submit = await page.getByRole('button', { name: 'Iniciar sesion' })
```
Comment (cerebrl) [inline tests/widget/inline/widget-inline.es.test.js:14]: I do like how this aligns the convention with Jest/Testing Library, but it is more verbose when we aren't actually using the variable afterwards. I see the `getBy*` convention better than `.locator`, just not sure it's a good replacement for when we just straight up want to trigger a user event, like `.fill`.

**Comment by ryanbas21** [comment]
@cerebrl thinking we can close this?

**Review by ryanbas21 — APPROVED** [review]

**Pull request merged**